runtime/signals.c

```diff
@@ -276,11 +276,12 @@ void caml_request_minor_gc (void)
 */
 
 CAMLno_tsan /* When called from [caml_record_signal], these memory
-               accesses may not be synchronized. */
-void caml_set_action_pending(void)
+               accesses may not be synchronized. Otherwise we assume
+               that we have unique access to dom_st. */
+void caml_set_action_pending(caml_domain_state * dom_st)
 {
-  Caml_state->action_pending = 1;
-  caml_interrupt_self();
+  dom_st->action_pending = 1;
+  atomic_store_rel(&dom_st->young_limit, (uintnat)dom_st->young_end);
```

Comment:
Is `young_limit == young_end` necessary here versus (uintnat) -1 that we use elsewhere?